pkg/apis/triggers/v1alpha1/trigger_binding_types.go

```diff
@@ -0,0 +1,63 @@
+package v1alpha1
+
+/*
+Copyright 2019 The Tekton Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// Check that TriggerBinding may be validated and defaulted.
+//var _ apis.Validatable = (*TriggerBinding)(nil)
+
+// +k8s:deepcopy-gen=true
+type TriggerBindingSpec struct {
+	Templates []TriggerTemplateRef `json:"templates,omitempty"`
+}
+
+// +k8s:deepcopy-gen=true
+type TriggerTemplateRef struct {
+	TemplateRef string  `json:"templateref,omitempty"`
+	Event       string  `json:"event,omitempty"`
+	Params      []Param `json:"params,omitempty"` // TODO: do we need default params?
```

Comment:
I think the instantiation of a Param will need a different type than the "ParamSpec" (thanks for the name @EliZucker 😉 )

supporting defaults makes sense - im wondering if it makes sense to use the types from Pipelines directly here to simplify the work we have to do

Reply:
Updated, hopefully correctly :D